pkg/skaffold/build/local/local.go

```diff
@@ -86,19 +87,19 @@ func (b *Builder) runBuildForArtifact(ctx context.Context, out io.Writer, a *lat
 
 	switch {
 	case a.DockerArtifact != nil:
-		return b.buildDocker(ctx, out, a, tag)
+		return b.buildDocker(ctx, out, a, ToDockerOpts(opts))
```

Comment:
It feels like this ImageOptions should be passed down into the builders.

Reply:
> Copied from above

Pushing ImageOptions into any of the actual builder implementations causes an import cycle (you had predicted this!). So I've defined ImageOptions as the generic builder-agnostic options object. The different builders define their own options object in their respective adapter package.